### PR TITLE
Docs update ReadTheDocs

### PR DESCRIPTION
**Pull request by og113**
Various updates to the docs, assuming we are going with ReadTheDocs.
- Moved to fully automatic API documentation (using `sphinx.ext.autosummary`), so that the docs are comprehensive and easier to maintain
- Added missing base example
- Updated installation
- Updated README.md with ReadTheDocs link
- Removed highlighting in LICENSE
- Updated code lines in First Example

Note that you can compare the docs made on this PR to those on main by comparing the following links:
- https://wallgo.readthedocs.io/en/docsupdatebeta/
- https://wallgo.readthedocs.io/en/latest/

### CODE REVIEW

**Comment by jorindevandevis** [comment]
Thanks Oli, this looks very good.

Just to be sure: the docs associated to this PR are in here, right https://wallgo.readthedocs.io/en/docsupdatebeta/ ? And if you merge this into main, they will also be here: https://wallgo.readthedocs.io/en/latest/ ?

When I go to the section Collected Examples, I read "[...] we have written an example base code, which the specific models utilise." and the description of the PR also writes that you added the base example, but I can not find it in the documentation.

**Comment by og113** [comment]
> Thanks Oli, this looks very good.
> 
> Just to be sure: the docs associated to this PR are in here, right https://wallgo.readthedocs.io/en/docsupdatebeta/ ? And if you merge this into main, they will also be here: https://wallgo.readthedocs.io/en/latest/ ?
> 
> When I go to the section Collected Examples, I read "[...] we have written an example base code, which the specific models utilise." and the description of the PR also writes that you added the base example, but I can not find it in the documentation.

Yes, your understanding of the merging process and associated docs pages is correct.

I have fixed the missing Collected Examples page, thanks for that. It was macOS being case-blind for filenames again, so that despite a case error the docs building worked locally but didn't work on the ReadTheDocs, which probably runs Linux and hence is not case-blind.

**Comment by og113** [comment]
Also, note that ReadTheDocs doesn't make docs for all the branches. I had to ask it specially to make the docs for the branch of this PR, and after the PR is merged I will remove https://wallgo.readthedocs.io/en/docsupdatebeta/.

**Review by jorindevandevis — APPROVED** [review]
It looks really good, thanks!

**Comment by jorindevandevis** [comment]
Maybe we have to wait with merging until the deleteSimpleModel branch is approved?

File: Models/Yukawa/yukawa.py [hunk 1]
```diff
@@ -228,180 +152,248 @@ def evaluate(
...
 
+## Don't run the main function if imported to another file
 if __name__ == "__main__":
-
-    example = YukawaModelExample()
-    example.runExample()
+    main()
+
+
+# class YukawaModelExample(WallGoExampleBase):
```
Comment (jorindevandevis) [inline Models/Yukawa/yukawa.py:223]: This should obviously be deleted, not just commented out. I will update it on my branch

**Comment by og113** [comment]
> Maybe we have to wait with merging until the deleteSimpleModel branch is approved?

We don't have the simple model in the list of examples, just the Yukawa model, so I think it should be okay.

**Pull request merged**